File: src/hack_forked/utils/core/build_id.ml
(*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *)

external get_build_revision : unit -> string = "hh_get_build_revision"

let build_revision = get_build_revision ()

File: rust_port/crates/flow_common/src/build_id.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#[cfg(fbcode_build)]
fn get_build_revision() -> &'static str {
    build_info::BuildInfo::get_revision()
}

#[cfg(not(fbcode_build))]
fn get_build_revision() -> &'static str {
    ""
}

pub fn build_revision() -> &'static str {
    get_build_revision()
}
